cluster-http/src/main/scala/akka/management/cluster/ClusterDomainEventServerSentEventEncoder.scala

```diff
@@ -0,0 +1,125 @@
+
+          case ClusterEvent.MemberExited(member) =>
+            sse("MemberExited", JsObject("member" -> encode(member)))
+
+          case ClusterEvent.MemberDowned(member) =>
+            sse("MemberDowned", JsObject("member" -> encode(member)))
+
+          case ClusterEvent.MemberRemoved(member, previousStatus) =>
+            sse("MemberRemoved", JsObject("member" -> encode(member), "previousStatus" -> encode(previousStatus)))
+        }
+
+      case leaderChanged: ClusterEvent.LeaderChanged =>
+        leaderChanged.leader match {
+          case Some(address) =>
+            sse("LeaderChanged", JsObject("address" -> encode(address)))
+
+          case None =>
+            sse("LeaderChanged", JsObject.empty)
+        }
+
+      case roleLeaderChanged: ClusterEvent.RoleLeaderChanged =>
+        roleLeaderChanged.leader match {
+          case Some(address) =>
+            sse(
+              "RoleLeaderChanged",
+              JsObject("role" -> JsString(roleLeaderChanged.role), "address" -> encode(address))
+            )
+
+          case None =>
+            sse("RoleLeaderChanged", JsObject("role" -> JsString(roleLeaderChanged.role)))
+        }
+
+      case ClusterEvent.ClusterShuttingDown =>
+        sse("ClusterShuttingDown", JsObject.empty)
+
+      case reachabilityEvent: ClusterEvent.ReachabilityEvent =>
+        reachabilityEvent match {
+          case ClusterEvent.UnreachableMember(member) =>
+            sse("UnreachableMember", JsObject("member" -> encode(member)))
+
+          case ClusterEvent.ReachableMember(member) =>
+            sse("ReachableMember", JsObject("member" -> encode(member)))
+        }
+
+      case dataCenterReachabilityEvent: ClusterEvent.DataCenterReachabilityEvent =>
+        dataCenterReachabilityEvent match {
+          case ClusterEvent.UnreachableDataCenter(dataCenter) =>
+            sse("UnreachableDataCenter", JsObject("dataCenter" -> JsString(dataCenter)))
+          case ClusterEvent.ReachableDataCenter(dataCenter) =>
+            sse("ReachableDataCenter", JsObject("dataCenter" -> JsString(dataCenter)))
+        }
+
+      case _ =>
+        // @TODO these are either internal, or added but forgot to update this module
+        // @TODO how should we handle them? maybe leave this as FIXME for now?
+
+        None
+    }
+  }
+
+  private def encode(address: Address): JsValue = JsString(address.toString)
+
+  private def encode(address: UniqueAddress): JsValue = JsObject(
+    "address" -> encode(address.address),
+    "longUid" -> JsNumber(address.longUid)
+  )
+
+  private def encode(memberStatus: MemberStatus): JsValue = JsString(
+    memberStatus match {
+      case MemberStatus.Joining  => "Joining"
+      case MemberStatus.WeaklyUp => "WeaklyUp"
+      case MemberStatus.Up       => "Up"
+      case MemberStatus.Leaving  => "Leaving"
+      case MemberStatus.Exiting  => "Exiting"
+      case MemberStatus.Down     => "Down"
+      case MemberStatus.Removed  => "Removed"
```

Comment:
maybe `case other => other.toString`?

Reply:
sounds good to handle that gracefully, another is coming soon https://github.com/akka/akka/pull/29838